src/resdk/tables/rna.py

```diff
@@ -216,12 +402,61 @@ def _cache_file(self, data_type: str) -> str:
         """Return full cache file path."""
         if data_type == self.META:
             version = self._metadata_version
+        elif data_type == self.QC:
+            version = self._qc_version
         else:
             version = self._data_version
 
         cache_file = f"{self.collection.slug}_{data_type}_{self.expression_source}_{self.expression_process_slug}_{version}.pickle"
         return os.path.join(self.cache_dir, cache_file)
 
+    def _download_qc(self) -> pd.DataFrame:
+        """Download sample QC data and transform into table.
+
+        QC info can be sourced from multiple files and multiple columns
+        in those files.
+        """
+        mqc_db = {}
+        for mqc in self.collection.data.filter(
+            type="data:multiqc",
+            status="OK",
+            ordering="created",
+            fields=["id", "name", "entity__id"],
+        ):
+            if mqc.sample is None:
+                continue
+            mqc_db[mqc.sample.id] = {
+                "mqc_id": mqc.id,
```

Comment:
Is it possible to have more than one `mqc` data object per sample? Probably not, but without this assumption the code does not work.

Reply:
The platform will not prevent you from re-running the MutiQC analysis on the same sample data, so it is possible to have multiple objects on the same sample. I suggest using the latest one in such cases, and reporting the warning to the user.

Reply:
Yes, this was exactly the reason why it was made as is. If there are multiple MultiQC Data in one Sample, only the latest will be used.